ProgrammaticControl/MainWindow.xaml.cs

```diff
@@ -221,8 +221,8 @@ public static Task RunAsync(this IExecutionService executionService)
                 }
             });
             executionService.MasterExecutable.CurrentSimpleExecutionStateChanged += viStateChangedEventHandler;
-            executionService.MasterExecutable.StartRun();
-            return result.Task;
+            await executionService.MasterExecutable.BeginRunAsync();

```

Comment:
`await RunAsync()` will return after the VI _starts_ running, so we still need to await the task that waits for it to finish running.